File: exp9_correlation.py
# ============================================================
# EXPERIMENT 9: Correlation & Multi-Variable Analysis
# ============================================================

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

print("=" * 50)
print("EXPERIMENT 9: Correlation & Multi-Variable Analysis")
print("=" * 50)

# ---------- Dataset ----------
np.random.seed(42)
n = 50
df = pd.DataFrame({
    'Age':         np.random.randint(22, 55, n),
    'Experience':  np.random.randint(1, 30, n),
    'Salary':      np.random.randint(30000, 100000, n),
    'Score':       np.random.randint(60, 100, n),
    'Hours_Work':  np.random.randint(30, 60, n)
})

print("\nDataset Head:\n", df.head())

# ---------- Correlation Matrix ----------
print("\n--- Correlation Matrix ---")
corr = df.corr()
print(corr.round(2))

# ---------- Heatmap ----------
plt.figure(figsize=(8, 6))
sns.heatmap(corr, annot=True, fmt=".2f", cmap='coolwarm',
            linewidths=0.5, square=True)
plt.title('Correlation Heatmap')
plt.tight_layout()
plt.savefig('exp9_heatmap.png')
plt.show()

# ---------- Scatter Plot Matrix (Pair Plot) ----------
sns.pairplot(df, diag_kind='kde', plot_kws={'alpha': 0.5})
plt.suptitle('Pair Plot - Multi-Variable Relationships', y=1.02)
plt.tight_layout()
plt.savefig('exp9_pairplot.png')
plt.show()

# ---------- Individual Scatter Plots ----------
fig, axes = plt.subplots(1, 3, figsize=(15, 5))

axes[0].scatter(df['Age'], df['Salary'], alpha=0.6, color='blue')
axes[0].set_title('Age vs Salary')
axes[0].set_xlabel('Age')
axes[0].set_ylabel('Salary')

axes[1].scatter(df['Experience'], df['Salary'], alpha=0.6, color='green')
axes[1].set_title('Experience vs Salary')
axes[1].set_xlabel('Experience')
axes[1].set_ylabel('Salary')

axes[2].scatter(df['Hours_Work'], df['Score'], alpha=0.6, color='red')
axes[2].set_title('Hours Worked vs Score')
axes[2].set_xlabel('Hours Work')
axes[2].set_ylabel('Score')

plt.tight_layout()
plt.savefig('exp9_scatter_plots.png')
plt.show()

# ---------- Grouped Bar Chart ----------
dept = pd.DataFrame({
    'Dept':   ['HR', 'IT', 'Finance', 'HR', 'IT', 'Finance'],
    'Salary': [45000, 70000, 60000, 40000, 75000, 55000],
    'Score':  [80, 90, 85, 75, 88, 82]
})
dept_avg = dept.groupby('Dept').mean()
dept_avg.plot(kind='bar', figsize=(7, 5), colormap='Set2', edgecolor='black')
plt.title('Average Salary & Score by Department')
plt.xlabel('Department')
plt.xticks(rotation=0)
plt.tight_layout()
plt.savefig('exp9_grouped_bar.png')
plt.show()

print("\nAll plots saved. Experiment 9 Complete!")

File: exp9_correlation.R
# ============================================================
# EXPERIMENT 9: Correlation & Multi-Variable Analysis in R
# Run this file directly in RStudio
# ============================================================

# ---------- Install if needed ----------
# install.packages(c("ggplot2", "corrplot", "GGally"))

cat("========================================\n")
cat("EXPERIMENT 9: Correlation & Multi-Variable Analysis\n")
cat("========================================\n")

# ---------- Dataset ----------
set.seed(42)
n <- 50
df <- data.frame(
  Age        = sample(22:55, n, replace=TRUE),
  Experience = sample(1:30,  n, replace=TRUE),
  Salary     = sample(30000:100000, n, replace=TRUE),
  Score      = sample(60:100, n, replace=TRUE),
  Hours_Work = sample(30:60,  n, replace=TRUE)
)

cat("\nDataset Head:\n")
print(head(df))

cat("\nSummary:\n")
print(summary(df))

# ---------- Correlation Matrix ----------
cat("\n--- Correlation Matrix ---\n")
corr_matrix <- cor(df)
print(round(corr_matrix, 2))

# ---------- corrplot - Heatmap ----------
library(corrplot)
cat("\n--- Correlation Heatmap (corrplot) ---\n")
corrplot(corr_matrix,
         method  = "color",
         addCoef.col = "black",
         tl.col  = "black",
         title   = "Correlation Heatmap",
         mar     = c(0,0,1,0))

# Circular style
corrplot(corr_matrix,
         method = "circle",
         type   = "upper",
         tl.col = "black",
         title  = "Correlation Circle Plot",
         mar    = c(0,0,1,0))

# ---------- Scatter Plots ----------
cat("\n--- Scatter Plots ---\n")
par(mfrow=c(1,3))

plot(df$Age, df$Salary,
     main="Age vs Salary",
     xlab="Age", ylab="Salary",
     col="blue", pch=16)
abline(lm(Salary ~ Age, data=df), col="red", lwd=2)

plot(df$Experience, df$Salary,
     main="Experience vs Salary",
     xlab="Experience", ylab="Salary",
     col="green", pch=16)
abline(lm(Salary ~ Experience, data=df), col="red", lwd=2)

plot(df$Hours_Work, df$Score,
     main="Hours Work vs Score",
     xlab="Hours Work", ylab="Score",
     col="purple", pch=16)
abline(lm(Score ~ Hours_Work, data=df), col="red", lwd=2)

par(mfrow=c(1,1))

# ---------- Pairs Plot (Scatter Matrix) ----------
cat("\n--- Pairs Plot ---\n")
pairs(df,
      main  = "Scatter Matrix - All Variables",
      col   = "steelblue",
      pch   = 16,
      cex   = 0.8)

# ---------- GGally pairplot (ggplot2 style) ----------
library(GGally)
ggpairs(df,
        title = "GGally Pair Plot") +
  theme_minimal()

# ---------- ggplot2 Visualizations ----------
library(ggplot2)
cat("\n--- ggplot2 Visualizations ---\n")

# Scatter with trend
p1 <- ggplot(df, aes(x=Age, y=Salary)) +
  geom_point(color="steelblue", size=2, alpha=0.7) +
  geom_smooth(method="lm", color="red", se=TRUE) +
  ggtitle("Age vs Salary with Regression Line") +
  theme_minimal()
print(p1)

# Multi-variable scatter
p2 <- ggplot(df, aes(x=Experience, y=Salary, size=Score, color=Hours_Work)) +
  geom_point(alpha=0.7) +
  scale_color_gradient(low="blue", high="red") +
  ggtitle("Experience vs Salary (Size=Score, Color=Hours)") +
  theme_minimal()
print(p2)

# ---------- Grouped Bar Chart ----------
cat("\n--- Department Analysis ---\n")
dept_df <- data.frame(
  Dept   = c("HR","IT","Finance","HR","IT","Finance"),
  Salary = c(45000,70000,60000,40000,75000,55000),
  Score  = c(80,90,85,75,88,82)
)
dept_avg <- aggregate(. ~ Dept, data=dept_df, FUN=mean)
cat("Department Averages:\n")
print(dept_avg)

# Reshape for grouped bar
dept_long <- tidyr::pivot_longer(dept_avg,
                                  cols=c(Salary, Score),
                                  names_to="Metric",
                                  values_to="Value")

p3 <- ggplot(dept_long, aes(x=Dept, y=Value, fill=Metric)) +
  geom_bar(stat="identity", position="dodge") +
  ggtitle("Average Salary & Score by Department") +
  theme_minimal()
print(p3)

# ---------- cor.test (significance) ----------
cat("\n--- Correlation Significance Test ---\n")
cat("Age vs Salary:\n")
print(cor.test(df$Age, df$Salary))

cat("\nExperience vs Salary:\n")
print(cor.test(df$Experience, df$Salary))

cat("\n========================================\n")
cat("Experiment 9 Complete!\n")
cat("========================================\n")
